src/AoC2024/1/aoc24_1.cpp

```diff
@@ -1,7 +1,66 @@
-#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <string>
 
-int main(){
-    std::cout << "Hello world!" << '\n';
+#include "aoc24_1.h"
 
-    return 0;
+std::vector<int>
+calculateDistances(const std::vector<std::pair<int,int>>& preparedList) {
+    std::vector<int> distances = {};
+
+    const bool listIsNotEmpty = preparedList.size() > 0;
+    if(listIsNotEmpty)
+        std::for_each(preparedList.begin(), preparedList.end(), [&](const std::pair<int,int>& pair){
+            int left = std::get<0>(pair);
+            int right = std::get<1>(pair);
+            int distance = std::abs(left - right);
+            distances.push_back(distance);
+        });
+    
+    return distances;
+}
+
+std::pair<std::vector<int>, std::vector<int>>
+readPairsFromFile(const std::filesystem::path& inputfile){
+    std::vector<int> left = {};
+    std::vector<int> right = {};
+    constexpr int expectedNumbeOfLines = 1000;
+    left.reserve(expectedNumbeOfLines);
+    right.reserve(expectedNumbeOfLines);
+
+    std::string line = {""};
+    int num1 = 0;
+    int num2 = 0;
+    std::ifstream file(inputfile);
+    if (!file.is_open())
+        throw std::runtime_error("Failed to open file");
+
+    while (std::getline(file, line)) {
+        std::istringstream iss(line);
+        if (!(iss >> num1 >> num2)) {
+            throw std::runtime_error("Invalid input format");
+        }
+        left.push_back(num1);
+        right.push_back(num2);
```

Comment:
I like to allocate the vector size in this case. I know it will contain 1000 elements and I'm loop-inserting to the vector that otherwise could (would?) result in a couple of reallocations